Test-№2.1-from-14.12.2022/task1.c

```diff
@@ -0,0 +1,197 @@
+        }
+        c = fgetc(file);
+    }
+    if (!(existCheck[0] && existCheck[1] && existCheck[2]))
+    {
+        return false;
+    }
+    return true;
+}
+
+int toInt(int *array)
+{
+    int answer = 0;
+    for (int i = 0; i < 8; ++i)
+    {
+        answer += array[i] * fastPower(10, i);
+    }
+    return answer;
+}
+
+long long orderCounter(FILE *file)
+{
+    long long preResult = 0;
+    long long result = 0;
+    int temp[8] = {0};
+    int counterOfItem = 0;
+    int priceOfItem = 0;
+    char c = fgetc(file);
+    int pointer = 0;
+    while (c != EOF)
+    {
+        if (pointer == 0)
+        {
+            while (c != '\t')
+            {
+                c = fgetc(file);
+            }
+        } else if (pointer == 1)
+        {
+            for (int i = 0; i < 8; ++i)
+            {
+                if (c == '\t' || c == '\n')
+                {
+                    break;
+                }
+                temp[i] = (c - '0');
+                c = fgetc(file);
+            }
+            if (c != '\t')
+            {
+                c = fgetc(file);
+                if (c != '\t')
+                {
+                    return -1;
+                }
+            }
+            counterOfItem = toInt(temp);
+            resetTemp(temp);
+        } else if (pointer == 2)
+        {
+            for (int i = 0; i < 8; ++i)
+            {
+                if (c == '\t' || c == '\n' || c == EOF)
+                {
+                    break;
+                }
+                temp[i] = (c - '0');
+                c = fgetc(file);
+            }
+            if (!(c == '\n' || c == EOF))
+            {
+                c = fgetc(file);
+                if (!(c == '\n' || c == EOF))
+                {
+                    return -1;
+                }
+            }
+            priceOfItem = toInt(temp);
+            resetTemp(temp);
+        }
+        if (c == '\t')
+        {
+            ++pointer;
+            c = fgetc(file);
+        } else if (c == '\n' || c == EOF)
+        {
+            pointer = 0;
+            preResult = priceOfItem * counterOfItem;
+            result += preResult;
+            priceOfItem = 0;
+            counterOfItem = 0;
+            preResult = 0;
+            c = fgetc(file);
+        }
+    }
+    return result;
+}
+
+int main()
+{
+    printf("Hello! I will count the price for all your\n");
+    printf("items in a file \"order.txt\". If your file\n");
+    printf("isn't correct, I will print message on your screen.\n\n");
+    FILE *fileTesting = fopen("order.txt", "r");
+    if (fileTesting == NULL)
+    {
+        printf("This file doesn't exist!\n");
+    }
+    if (!isFileTabulationCorrect(fileTesting))
+    {
+        printf("Your file is not correct!\n");
+        printf("Please, repair file's tabulations and line breaks!\n");
+        fclose(fileTesting);
+        return -1;
+    }
+    fclose(fileTesting);
+    FILE *file = fopen("order.txt", "r");
```

Comment:
```suggestion
    fseek(fileTesting, 0, 0);
```
Так можно переместить указатель в начало файла. Тогда можно не переоткрывать файл.
А если закрываешь и открываешь файл, то нужно снова проверить, существует ли файл (вдруг его удалят в это самый момент, когда он уже закрыт, но еще не открыт). Такого рода баги редко проявляются, ноэтому их сложно отлаживать.